content/en/blog/2024/prom-and-otel/index.md

```diff
@@ -0,0 +1,671 @@
+  - apiGroups: ['']
+    resources:
+      - configmaps
+    verbs: ['get']
+  - apiGroups:
+      - discovery.k8s.io
+    resources:
+      - endpointslices
+    verbs: ['get', 'list', 'watch']
+  - apiGroups:
+      - networking.k8s.io
+    resources:
+      - ingresses
+    verbs: ['get', 'list', 'watch']
+  - nonResourceURLs: ['/metrics']
+    verbs: ['get']
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: opentelemetry-targetallocator-rb
+subjects:
+  - kind: ServiceAccount
+    name: opentelemetry-targetallocator-sa
+    namespace: opentelemetry
+roleRef:
+  kind: ClusterRole
+  name: opentelemetry-targetallocator-role
+  apiGroup: rbac.authorization.k8s.io
+```
+
+Zooming in a bit on the previous
+[ClusterRole](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole),
+the following rules will provide the minimum access required for the Target
+Allocator to query all the targets it needs based on any Prometheus
+configurations:
+
+```yaml
+- apiGroups: ['']
+  resources:
+    - nodes
+    - nodes/metrics
+    - services
+    - endpoints
+    - pods
+  verbs: ['get', 'list', 'watch']
+- apiGroups: ['']
+  resources:
+    - configmaps
+  verbs: ['get']
+- apiGroups:
+    - discovery.k8s.io
+  resources:
+    - endpointslices
+  verbs: ['get', 'list', 'watch']
+- apiGroups:
+    - networking.k8s.io
+  resources:
+    - ingresses
+  verbs: ['get', 'list', 'watch']
+- nonResourceURLs: ['/metrics']
+  verbs: ['get']
+```
+
+If you enable the `prometheusCR` (set
+`spec.targetAllocator.prometheusCR.enabled` to `true`) in the
+OpenTelemetryCollector CR, you will also need to define the following roles.
+These give the Target Allocator access to the PodMonitor and ServiceMonitor CRs.
+It also gives namespace access to the PodMonitor and ServiceMonitor.
+
+```yaml
+- apiGroups:
+    - monitoring.coreos.com
+  resources:
+    - servicemonitors
+    - podmonitors
+  verbs:
+    - '*'
+- apiGroups: ['']
+  resources:
+    - namespaces
+  verbs: ['get', 'list', 'watch']
+```
+
+## Additional OTel components for Kubernetes
+
+This section covers additional OTel Collector components you can use to capture
+Kubernetes metrics.
+
+Receiving data:
+
+- [Kubernetes Cluster Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8sclusterreceiver):
+  collects cluster-level metrics and entity events from the
+  [Kubernetes API server](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/)
+- [Kubernetes Objects Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8sobjectsreceiver):
+  collects (pull/watch) objects from the Kubernetes API server
+- [Kubelet Stats Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver):
+  pulls node, pod, container, and volume metrics from the API server on a
```

Comment:
I would just say it pulls metrics from the kubelet.  API server in k8s usually refers to something serving k8s resources, which the kubelet does not.